### ISSUE TITLE
Add userId argument to "groups.list" REST API call

### ISSUE THREAD

**Issue by mbrodala** [RocketChat/Rocket.Chat#12943]
The [`groups.list` API call](https://rocket.chat/docs/developer-guides/rest-api/groups/list/) lists all private groups of the current user.

It would be very useful to retrieve the list of private groups for a given user, similar to the [`groups.counters` API call](https://rocket.chat/docs/developer-guides/rest-api/groups/counters/). This could be achieved similarly by adding a `userId` argument.

**Comment by mbrodala** [comment]
Notice that #12029 requested something similar.

**Comment by MarcosSpessatto** [comment]
@mbrodala Actually you can't retrieve the private groups of another user. You can only retrieve the groups that you are part of.

**Comment by mbrodala** [comment]
Correct, and with this issue I'd like to request an additional parameter to change that. Or are there some deeper architectural decisions which would prevent this?

**Comment by simonfoxguitar** [comment]
Is it correct that there is no way for an Admin to tell which private groups a given user belongs to?
I tried users.info with the fields={"userRooms":1} setting, but no luck there.

**Comment by mbrodala** [comment]
@simonfoxguitar Correct, right now I need to use this workaround:

1. Use [`groups.list`](https://rocket.chat/docs/developer-guides/rest-api/groups/list/) to get all private groups of an instance.
2. Iterate these groups and use [`groups.members`](https://rocket.chat/docs/developer-guides/rest-api/groups/members/) to get all members of each private group.
3. Iterate these members and compare their `_id` with the one for the given user. If you find a match, you have one more group a user belongs to.

I know it's cumbersome and requires quite a few API calls but it's the only way I know of.

**Comment by engelgabriel** [comment]
Can you explain better the use case here? This is indeed a privacy discussion more than an architectural decision. We would like to move this discussion to the forum to get input from the broader community.

**Comment by engelgabriel** [comment]
We don't want platform admins to have the same access to the database, as the DB admins, unless they are the same person. So one idea would be to have a flag, that can only be set directly on the DB to give this kind of access via our admin panel.

**Comment by mbrodala** [comment]
Not sure how this would help here. Our usecase is retrieving the groups/channels a user belongs in RC and displaying this in a separate system. Right now we use a dedicated admin/bot user for that and it would be useful if that user had full access to everything.

**Comment by awunder** [comment]
my use case is to remove / add a user from / to groups based on his subscription status an a paid member site..

i need to be able to quickly determine in which groups a user is in without looping through all groups

**Comment by seregasoft** [comment]
Hi @mbrodala ! Please tell me how you get members by "groups.members"? I have a issue https://github.com/RocketChat/Rocket.Chat/issues/22865

groups.list response fine, but groups.members fails remotely and locally with a "room-not-found" unless user of token that i use via API are also in this group. Role of this user is Admin with all checkboxes in Permission settings

> @simonfoxguitar Correct, right now I need to use this workaround:
> 
> 1. Use [`groups.list`](https://rocket.chat/docs/developer-guides/rest-api/groups/list/) to get all private groups of an instance.
> 2. Iterate these groups and use [`groups.members`](https://rocket.chat/docs/developer-guides/rest-api/groups/members/) to get all members of each private group.
> 3. Iterate these members and compare their `_id` with the one for the given user. If you find a match, you have one more group a user belongs to.
> 
> I know it's cumbersome and requires quite a few API calls but it's the only way I know of.

**Comment by mbrodala** [comment]
Sorry, I'm not working on the project with RocketChat integration anymore so I cannot give you useful pointers here.
